src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -185,10 +185,10 @@ private String executeCommand(String commandText) throws CommandException, Parse
             throw e;
         }
 
-        if (commandResult.equals(Messages.MESSAGE_SHOWING_HELP)) {
+        if (commandResult == Messages.MESSAGE_SHOWING_HELP) {
             handleHelp();
         }
-        if (commandResult.equals(Messages.MESSAGE_EXIT_ACKNOWLEDGEMENT)) {
+        if (commandResult == Messages.MESSAGE_EXITING) {
```

Comment:
Should string comparison should be using `.equals()` instead of `==`?

Reply:
It should be `==` so that it will only enter to if block when it is the exact same object.
This is to prevent an edge case where a command may somehow return this exact string, but is not actually a HELP or EXIT command.